subprojects/core-api/src/main/java/org/gradle/BuildListener.java

```diff
@@ -26,7 +27,7 @@
  * @see org.gradle.api.invocation.Gradle#addListener(Object)
  */
 @EventScope(Scopes.Build.class)
-public interface BuildListener {
+public interface BuildListener extends ListenerDeprecatedInGradleScope {
```

Comment:
I wonder if an annotation would be better.

Reply:
Replaced with an annotation, it has benefit of being internal.